internal/pkg/ctxattr/ctxattr_test.go

```diff
@@ -0,0 +1,90 @@
+package ctxattr
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	"go.uber.org/zap"
+)
+
+func TestContextAttributes(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	set := Attributes(ctx)
+	_, ok := set.Value("isValid")
+	assert.False(t, ok)
+	assert.Equal(t, 0, set.Len())
+
+	fields := ZapFields(ctx)
+	assert.Empty(t, fields)
+
+	ctx = ContextWith(ctx, attribute.Bool("isValid", true), attribute.String("status", "success"))
+	ctx = ContextWith(ctx, attribute.Bool("isValid", false), attribute.Int("count", 5))
+
+	set = Attributes(ctx)
+	assert.Equal(t, 3, set.Len())
+
+	value, ok := set.Value("isValid")
+	require.True(t, ok)
+	assert.Equal(t, "false", value.Emit())
+
+	value, ok = set.Value("status")
+	require.True(t, ok)
+	assert.Equal(t, "success", value.Emit())
+
+	value, ok = set.Value("count")
+	require.True(t, ok)
+	assert.Equal(t, "5", value.Emit())
+
+	fields = ZapFields(ctx)
+	assert.Len(t, fields, 3)
+
+	assert.True(t, fields[0].Equals(zap.Int64("count", 5)))
+	assert.True(t, fields[1].Equals(zap.Bool("isValid", false)))
+	assert.True(t, fields[2].Equals(zap.String("status", "success")))
+}
+
+func TestAttributeConversion(t *testing.T) {
+	t.Parallel()
+
+	assert.Equal(t, zap.Bool("isValid", true), convertAttributeToZapField(attribute.Bool("isValid", true)))
+	assert.Equal(t, zap.Int("count", 5), convertAttributeToZapField(attribute.Int("count", 5)))
+	assert.Equal(t, zap.Float64("result", 3.14), convertAttributeToZapField(attribute.Float64("result", 3.14)))
+	assert.Equal(t, zap.String("result", "success"), convertAttributeToZapField(attribute.String("result", "success")))
```

Comment:
Table driven test would be better for reading, checking and editing.

```
t.Parallel()

	cases := []struct {
		Name     string
		In       attribute.KeyValue
		Expected zap.Field
	}{
		{
			Name:     "bool",
			In:       attribute.Bool("isValid", true),
			Expected: zap.Bool("isValid", true),
		},
		{
			Name:     "int",
			In:       attribute.Int("count", 5),
			Expected: zap.Int("count", 5),
		},
		{
			Name:     "float64",
			In:       attribute.Float64("result", 3.14),
			Expected: zap.Float64("result", 3.14),
		},
		{
			Name:     "string",
			In:       attribute.String("result", "success"),
			Expected: zap.String("result", "success"),
		},
		{
			Name:     "booleans",
			In:       attribute.BoolSlice("booleans", []bool{true, false}),
			Expected: zap.Bools("booleans", []bool{true, false}),
		},
		{
			Name:     "integers",
			In:       attribute.IntSlice("integers", []int{0, 5, -10}),
			Expected: zap.Int64s("integers", []int64{0, 5, -10}),
		},
		{
			Name:     "floats",
			In:       attribute.Float64Slice("floats", []float64{0.0, 5.4, -10.7}),
			Expected: zap.Float64s("floats", []float64{0.0, 5.4, -10.7}),
		},
		{
			Name:     "strings",
			In:       attribute.StringSlice("strings", []string{"success", "error"}),
			Expected: zap.Strings("strings", []string{"success", "error"}),
		},
	}

	for _, tc := range cases {
		assert.Equal(t, tc.Expected, convertAttributeToZapField(tc.In), tc.Name)
	}
```

------------

- `name` is optional, you can use test case index in the assert error message.
- you can simply convert it to parallel sub tests (not needed here)
```
	for _, tc := range cases {
		t.Run(tc.Name, func(t *testing.T) {
		        t.Parallel()
			assert.Equal(t, tc.Expected, convertAttributeToZapField(tc.In))	
		})
	}
```